### PR TITLE
Replace tm-db dependency with store package

### LINKED ISSUE #269: Remove tm-db dependency from indexer
Indexer code was borrowed from Tendermint. It's using tm-db to access KV store.
This generates two problems:
 - we can't use the same KV store instance for optimint and indexer (this is not elegant, wasteful, etc)
 - it seems there is some global state in tm-db. when optimint is used in ethermint, there is a conflict causing problems at execution.

### PR DESCRIPTION
**Pull request by tzdybal**
* added prefix iterator support to `store`
* replaced tm-db dependency with `store` package in indexer
* bonus: indexer now use the same KV store instance as rest of optimint

Resolves #269

### CODE REVIEW

**Comment by tzdybal** [comment]
@jbowen93 this PR should unblock you on indexer/tm-db issue.

File: go.mod
Comment (liamsi) [inline go.mod:158]: just noting that tm-db is still a dep in case that was causing issues (I remember some slack discussions around this with @jbowen93).

**Review by liamsi — COMMENTED** [review]
OK, so I don't fully have the context of the changes here but they make sense and the PR looks good to me.

Questions/comments before I approve this:

 - Please open an issue describing why this change make sense.
 - "indexer now use the same KV store instance as rest of optimint" could this have any downsides? Does tendermint also use the same DB instance for the indexer as the rest? 
 - Can @jbowen93 confirm that this unblocks him indeed?

**Review by liamsi — APPROVED** [review]

**Comment by jbowen93** [comment]
I can confirm this unblocks me here!

File: go.mod
Comment (jbowen93) [reply go.mod:158]: This is fine.

**Review by jbowen93 — APPROVED** [review]
Tested ACK

**Comment by tzdybal** [comment]
> * "indexer now use the same KV store instance as rest of optimint" could this have any downsides? Does tendermint also use the same DB instance for the indexer as the rest?

Separate database is easier to remove, if needed.  Tendermint is leaning into removing indexing code from codebase completely, as "it's not the nodes job to index externally used data".

**Pull request merged**
